### ISSUE TITLE
KV does not respect "prefix" when testing KV store connection 

### ISSUE THREAD

**Issue by vincentmele** [containous/traefik#2394]
### Do you want to request a *feature* or report a *bug*?
Report a bug.

### What did you do?
Used an ACL in Consul 1.0.0 to restrict Traefik to read/write access "traefik" key/folder. [consul] "prefix" is set to "traefik".

### What did you expect to see?
I expected Traefik to only need write access to the key/folder specified in [consul] "prefix" .

### What did you see instead?
Traefik failed to start with an error message in the logs of "KV connection error: Failed to test KV store connection: Unexpected response code 403."

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.4.2
Codename:     roquefort
Go version:   go1.9.2
Built:        2017-11-02_01:47:46PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
Using consul 1.0.0 in a separate docker stack, with a shared network between traefik and consul.

traefik docker-compose.yml snippet:
```yaml
version: '3.4'
services:
  traefik:
    image: traefik
    command: -c /dev/null --web --docker --docker.domain=docker.localhost --logLevel=DEBUG --consul --consul.endpoint=consul_server:8500
```

Working ACL (as a workaround):
```
key "traefik" {
  policy = "write"
},
key "qmslkjdfmqlskdjfmqlksjazçueznbvbwzlkajzebvkwjdcqmlsfj" {
  policy = "write"
}
```
Working ACL (full root access; not desirable):
```
key "" {
  policy = "write"
}
```
Non-Working and (desirable) ACL:
```
key "traefik" {
  policy = "write"
}
```
Using an ACL with full read/write access or read/write access to the specific test key (found in the source code) allows traefik to fully work as expected.

This bug resides on line 105 of provider/kv/kv.go https://github.com/containous/traefik/blob/7192aa86b5264e014ceabceb6367d8e6ce9a2fe6/provider/kv/kv.go#L105

Instead of writing and checking the test key within the "prefix" folder, traefik attempts to write and read it from the root. This is not desirable from a security standpoint, is outside of the expected behavior, and is not documented.

I don't have a build environment up and running yet, but may contribute a pull to fix this bug at a later time.

**Comment by vincentmele** [comment]
Additionally, when running Traefik in Docker swarm mode, Traefik will not elect a leader without additional permissions. After trial and error, I discovered that the "session" Consul permission is needed.
Therefore, my "working" ACL is:
```json
key "traefik" {
  policy = "write"
},
key "qmslkjdfmqlskdjfmqlksjazçueznbvbwzlkajzebvkwjdcqmlsfj" {
  policy = "write"
},
session "" {
  policy = "write"
}

Please document these needed permissions. I suggest in the HA/Clustering and/or Consul documentation

**Comment by nmengin** [comment]
@vincentmele Many thanks for the analyzis!

In fact, for the moment, you can just allow `read access` for the key `qmslkjdfmqlskdjfmqlksjazçueznbvbwzlkajzebvkwjdcqmlsfj` but it's quite ugly...

I will fix this point with a PR to add the `traefik` prefix to this _healthcheck_ key and complete the documentation.

**Comment by traefiker** [comment]
Closed by #2400.
